src/tests/test_raft_repl_dev.cpp

```diff
@@ -704,6 +853,66 @@ TEST_F(RaftReplDevTest, GCReplReqs) {
 }
 #endif
 
+TEST_F(RaftReplDevTest, BaselineTest) {
+    // Testing the baseline resync where leader creates snapshot and truncate entries.
+    // To simulate that write 10 entries to leader. Restart follower 1 with sleep 20s.
+    // Write to leader again to create 10 additional entries which follower 1 doesnt have.
+    // This is the baseline data. Truncate and snapshot on leader. Wait for commit for leader
+    // and follower 2. Write to leader again 10 entries after snapshot to create entries
+    // for incremental resync. We can create snapshot manually or triggered by raft.
+    // Verify all nodes got 20 entries.
+    LOGINFO("Homestore replica={} setup completed", g_helper->replica_num());
+    g_helper->sync_for_test_start();
+
+#ifdef _PRERELEASE
+    // If debug build we set flip to force truncate.
+    if (g_helper->replica_num() == 0) {
+        LOGINFO("Set force home logstore truncate");
+        g_helper->set_basic_flip("force_home_raft_log_truncate");
+    }
+#endif
+
+    // Write on leader.
+    uint64_t entries_per_attempt = 10;
+    LOGINFO("Write on leader num_entries={}", entries_per_attempt);
+    this->write_on_leader(entries_per_attempt, true /* wait_for_commit */);
+
+    // Restart follower-1 with delay.
+    this->restart_replica(1, 20 /* shutdown_delay_sec */);
+
+    LOGINFO("Write on leader num_entries={}", entries_per_attempt);
+    this->write_on_leader(entries_per_attempt, true /* wait_for_commit */);
+
+    if (g_helper->replica_num() == 0 || g_helper->replica_num() == 2) {
+        // Wait for commmit on leader and follower-2
+        this->wait_for_all_commits();
+        LOGINFO("Got all commits for replica 0 and 2");
+    }
+
+    if (g_helper->replica_num() == 0) {
+        // Leader does manual snapshot and truncate
+        LOGINFO("Leader create snapshot and truncate");
+        this->create_snapshot();
+        this->truncate(0);
+    }
```

Comment:
Should we just write the `snapshot_freq_distance` number of logs to cause an internal snapshot like we'll experience in a real setup? We never call `snapshot` directly from above the library from what I know.

Reply:
We do set snapshot_freq_distance at line 946.